rudder-web/src/main/webapp/javascript/rudder/homePage.js

```diff
@@ -166,9 +168,86 @@ function doughnutChart (id,data,count, colors) {
   }
   
   var chart = new Chart(context, chartOptions);
+  window[id] = chart;
   context.after(chart.generateLegend())
 }
 
+
+// home page chart legend function
+
+// Hide data on click
+ function hideChartData (event, index, name) {
+  // Get chart data
+  var chart = event.view[name];
+  var meta = chart.getDatasetMeta(0);
+  // Hide selected data
+  meta.data[index].hidden = !meta.data[index].hidden;
+  chart.update();
+};
+
+// When hovering legend display tooltip and highlight
+function openTooltip(event ,index, name){
+  // Get chart
+  var chart = event.view[name];
+  
+  // Check if the element is already in the 'active' elements of the chart
+  if(chart.tooltip._active == undefined)
+     chart.tooltip._active = []
+  var activeElements = chart.tooltip._active;
+  
+  // Get our element
+  var requestedElem = chart.getDatasetMeta(0).data[index];
+  
+  // If already in active elements, skip
+  for(var i = 0; i < activeElements.length; i++) {
+      if(requestedElem._index == activeElements[i]._index)  
+         return;
+  }
+  
+  // Highlight element
+  requestedElem.custom = requestedElem.custom || {};
+  requestedElem.custom.backgroundColor = Chart.helpers.getHoverColor(requestedElem._view.backgroundColor)
+  
+  // Add element
+  activeElements.push(requestedElem);
+  chart.tooltip._active = activeElements;
+  
+  chart.tooltip.update(true);
+  chart.update();
+  chart.update(0);
+  chart.draw();
+}
+
+//When mouse out of legend remove tooltip and highlight
+function closeTooltip(e ,index, name){
+  // Get chart
+  var chart = e.view[name];
+  
+  // Get active elements
+  var activeElements = chart.tooltip._active;
+  if(activeElements == undefined || activeElements.length == 0)
+    // No elements, nothing to do
+    return;
```

Comment:
I didn't know js allowed if without {} and statement not on the immediate next line